futures-executor/src/local_pool.rs

```diff
@@ -157,20 +170,94 @@ impl LocalPool {
         })
     }
 
+    /// Runs all tasks and returns after completing one future or until no more progress
```

Comment:
In Fuchsia we call this function `run_until_stalled`. Maybe that would work here as well?